bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -73,6 +74,8 @@ class CheckRun {
 
     private Duration expiresIn;
 
+    private Optional<JdkVersion> versionOpt;
```

Comment:
Fields should not be `Optional`. We should also not add fields that are arbitrarily initiated as a side effect of some method. This needs to be initiated in one place and then propagated through parameters. Looking at the calling tree of the various getIssue methods, this looks like a pretty bad mess. Not sure how to untangle this to propagate the version value.

Reply:
I add this field because I don't want to call `BotUtils.getVersion(pr)` twice. If adding a field is a bad idea, I think just call `BotUtils.getVersion(pr)` twice is better than add `JdkVersion` to many method interfaces.

Reply:
I think the calling tree of `issues()` is in a mess because we have two different type of issues(`org.openjdk.skara.vcs.openjdk` and `org.openjdk.skara.issuetracker.Issue`). Besides,  we don't have class fields to store issues, csrIssues, JepIssues. So we need to call `issues()` three times.

Reply:
> Yes, the two different Issue types definitely adds to the confusion, but we also have the `::issues` method that takes two booleans. It has three callers and only one of those sets the parameters true. This should just have been split into different methods. As it is now, they are calling each other in a circle, only saved by the boolean being false in the second call. Untangling this would make it pretty straightforward to provide a JdkVersion argument to the method for getting just CSR issues.
> 
> I think there is value in only calling `BotUtils.getVersion(pr)` once as it's quite expensive to evaluate.

Good idea, I will split them into different methods. Thx!

Reply:
I started looking at this call chain some more. We are already fetching issues multiple times, that's really what's causing this problem in the first place. We should only fetch issues once in a CheckRun. It will take some refactoring, since we need to preserve the IssueTracker Issues for CSR issues, but still worth it I think.

Reply:
> I started looking at this call chain some more. We are already fetching issues multiple times, that's really what's causing this problem in the first place. We should only fetch issues once in a CheckRun. It will take some refactoring, since we need to preserve the IssueTracker Issues for CSR issues, but still worth it I think.
> 
> So instead of sending the fixVersion around, we should be fetching issues at a higher level and send those around.

I think so too. So I already did some refactoring, now we will only fetch issues once and fetch csr issues once too.